app/messages/validators/security_group_rule_validator.rb

```diff
@@ -131,34 +114,42 @@ def valid_destination_type(destination, record, index)
   end
 
   def validate_destination(destination, record, index)
-    address_list = destination.split('-')
     error_message = 'destination must be a valid CIDR, IP address, or IP address range'
 
+    comma_delimited_destinations_enabled = CloudController::RuleValidator.comma_delimited_destinations_enabled?
+    error_message = 'destination must contain valid CIDR(s), IP address(es), or IP address rang(es)' if comma_delimited_destinations_enabled
+
+    unless destination.index(',').nil?
+      unless comma_delimited_destinations_enabled
+        add_rule_error(error_message, record, index)
+        return
+      end
+
+      destinations = destination.partition(',')
+      destination = destinations.first
+      remainder = destinations.last
+
+      validate_destination(remainder, record, index)
```

Comment:
I personally think using recursion here is less clear than doing something like `destination.split(',').each { |foo| validate_destination(foo)}` up on line 30

Reply:
I tried implementing this with iteration instead of recursion in this commit:
- https://github.com/cloudfoundry/cloud_controller_ng/pull/3644/commits/c5c437db1546823259b4153aa970d6895002ddaa

Please let me know if you think that is more legible.

Reply:
That looks good to me.  Thanks for changing it!